### DISCUSSION TITLE
Jupytext in combination with launch buttons

### DISCUSSION THREAD

**Discussion by nilsleh** [executablebooks/meta#1109] [Q&A]
We are using the `sphinx_book_theme` for a project documentation because it has such a nice clean look and cool features. Especially, the launch button for tutorials is super handy. For version control purposes we recently explored `jupytext` as a way to write the tutorials with better version control but still have nicely rendered notebooks in the documentation. However, I noticed that since switching to jupytext appraoch, the "launch buttons" do not appear anymore and I am not clear how to configure this correctly. On your [documentation page](https://sphinx-book-theme.readthedocs.io/en/stable/content/launch.html) you state that binder automatically point to the .ipynb file instead of the .py file. However, if we have both of those files in the tutorials directory, then building the docs complains that `WARNING: multiple files found for the document`. 

As a reference, this is the [conf.py file](https://github.com/lightning-uq-box/lightning-uq-box/blob/main/docs/conf.py) and this is an [example tutorial](https://lightning-uq-box.readthedocs.io/en/latest/tutorials/regression/mc_dropout.html) on the documentations page. Would be grateful for any pointers as to how to configure this correctly, or whether this is even possible!

**Comment by nilsleh** [comment]
I found this issue from 2020: https://github.com/executablebooks/jupyter-book/issues/752 but I guess this is still the status quo

**Comment by agoose77** [comment]
You could try excluding the `.md` files from Sphinx, such that Sphinx thinks that there is only one document with the given name. You can use the top-level `exclude_patterns` in `_config.yml` to control this.

**Reply by nilsleh** [reply]
Thanks for your reply. I think in that case I would still need to keep track of both the .py files for jupytext and always generate .ipynb notebooks and also commit them to have the launch button available. But then one looses the version control benefit of jupytext files because one always needs to update and commit the .ipynb notebooks as well. I was hoping that maybe since .ipynb files are generated when the docs are built from the jupytext .py files, the launch button could be pointed towards those files, so you have both the version control benefit of just python files but the powerful launch button which is such a beneficial feature for tutorials :)
